compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2713,6 +2714,46 @@ private void checkConstantAssignment(BLangExpression varRef, AnalyzerData data)
         }
     }
 
+    private boolean checkIfFiniteTypeHasNullMembers(BType bType) {
+        if (bType.tag != TypeTags.FINITE) {
+            return false;
+        }
```

Comment:
If we write the type as a type definition, we are still getting a "variable assignment is required" error
Example
```ballerina
function vlaue() {
    f1();

}

type Foo null;
function f1() returns Foo => null;
```